src/porepy/numerics/ad/_ad_utils.py

```diff
@@ -294,6 +294,60 @@ def discretize_from_list(
                     pass
 
 
+def _validate_indices(
+    time_step_index: Optional[int] = None,
+    iterate_index: Optional[int] = None,
+) -> list[tuple[Any, int]]:
+    """Helper method to validate the indexation of getter and setter methods for
+    values in a grid's data dictionary.
+
+    See :func:`set_solution_values` and :func:`get_solution_values`.
+
+    """
+    if time_step_index is None and iterate_index is None:
+        raise ValueError(
+            "At least one of time_step_index and iterate_index needs to be different"
+            " from None."
+        )
+
+    out = []
+
+    if iterate_index is not None:
+        # Some iterate values of the current time
+        if iterate_index >= 0:
+            out.append((pp.ITERATE_SOLUTIONS, iterate_index))
+        # Negative iterate indices are not supported
+        else:
+            raise ValueError(
+                "Use increasing, non-negative integers for (previous) iterate values."
+            )
+
+    if time_step_index is not None:
+        # Some previous time.
+        if time_step_index > 0:
+            out.append((pp.TIME_STEP_SOLUTIONS, time_step_index))
+        # Current time. NOTE this is ambigous since the current time is an unknown and
+        # has multiple iterate values.
+        # Alternatively, we could associate time_step_index = 0 with iterate_index = 0
+        # the below elif branch introduces the convention that
+        # time step = iterate step = 0 are equivalent.
+        elif time_step_index == 0:
```

Comment:
@keileg I unified the time step indexation with the iterate indexation.
0 is now the current time and every time-dependent operator starts with that value.
Though current time (unknown value) is ambiguous when referring to it using time step indices.
I added respective value error to alert the user to use iterate_index instead.

Reply:
I agree, this is a fair state to leave it for now.